### PR TITLE
Prefill fields in PurchasePayments when generated through a Purchase

### LINKED ISSUE #1943: Some fields in outgoing payments are no longer pre-filled.
When you add a new purchase-payment from a purchase, some fields are no longer pre-filled : 
the 2 dates and the manager.

### PR DESCRIPTION
**Pull request by Aquaj**
Fixes #1943.
- Added a way to pass prefill options to the links in the new_x buttons in `affair_of` helper.
- Prefilled `responsible`, `paid_at` and `to_bank_at` in new PurchasePayment.

### CODE REVIEW

**Comment by EmmaDorie** [comment]
Please, could someone finish correcting this bug????

File: app/views/backend/purchases/show.html.haml [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
   - if resource.affair
     - c.cobble :affair do
-      = affair_of(resource, default: 'PurchasePayment')
+      = affair_of(resource, default: 'PurchasePayment', purchase_payment_attributes: { responsible_id: resource.responsible_id, paid_at: resource.payment_at, to_bank_at: resource.payment_at})
```
Comment (burisu) [inline app/views/backend/purchases/show.html.haml]: Why the option name is `<model>_attributes` and not `params` or `url_options` or something more "URLesque"

Comment (Aquaj) [inline app/views/backend/purchases/show.html.haml]: Do you mean something like `params: { <model>: { ... } }`? Or simply `<model>_params` or `<model>_url_options` ?
I would agree indeed for a change towards the latter but we still need a way to characterize these options by model as we might want to pass different options for each of the involved models in `affair_of`.

Maybe `url_options: { <model>: { ... }, <other_model>: { ... } }` ?

Comment (burisu) [reply app/views/backend/purchases/show.html.haml]: Yes, sorry, I forgot at that time that we can have other types... The most explicit name as parameter is: `new_<model>_url_options`. But your last proposal seems more sustainable.
Do you agree?

Comment (Aquaj) [reply app/views/backend/purchases/show.html.haml]: I agree. I will implement the last proposal.

**Pull request merged**